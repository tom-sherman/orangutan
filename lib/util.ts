export type Range = IterableIterator<number>
export type RangeFactory = () => Range

/**
 * Converts an array to a generator.
 */
export const arrToRangeFactory = (array: number[]): RangeFactory =>
  array[Symbol.iterator].bind(array)

/**
 * Converts an array to an iterator.
 */
export const arrToRange = (array: number[]): Range => arrToRangeFactory(array)()
