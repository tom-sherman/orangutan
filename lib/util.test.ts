import { arrToRangeFactory, arrToRange } from './util'

describe('util', () => {
  test('array to factory', () => {
    const factory = arrToRangeFactory([1, 2, 3])
    expect(factory).toBeInstanceOf(Function)
    expect(factory()).toMatchSnapshot()
  })

  test('array to range', () => {
    const r = arrToRange([1, 2, 3])
    expect(r.next()).toEqual({ value: 1, done: false })
    expect(r.next()).toEqual({ value: 2, done: false })
    expect(r.next()).toEqual({ value: 3, done: false })
    expect(r.next()).toEqual({ value: undefined, done: true })
  })
})
