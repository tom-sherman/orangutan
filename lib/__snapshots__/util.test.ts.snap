// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`util array to factory 1`] = `Object {}`;
